edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesEventLogger.cs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
+{
+    using System;
+    using k8s;
+    using k8s.Models;
+    using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class KubernetesEventLogger<T>
```

Comment:
>KubernetesEventLogger [](start = 17, length = 21)

The general pattern in this project is to have an Events subclass on each object that generates logs, rather than one Events logger per project.

Reply:
Since this takes an object as it's type, how is this different performing vs having a subclass that we would have to maintain per object?